### PR TITLE
Fix typos in docs

### PR DESCRIPTION
**Pull request by AD4MANTIS**
I just used your crate for my own project an corrected some typos while reading the docs and book.
I'm not a native english speaker myself so my edit may not be 100% correct.

If you are using VS Code i suggest using the extension `streetsidesoftware.code-spell-checker` & `streetsidesoftware.code-spell-checker-french`.
with the setting `cSpell.language` you can also set multiple languages like `en,fr`.

- [x] docs/book/src/declare/03_plurals.md:100 i think the comparison was the wrong way

I also added an extra test on top 😊

### CODE REVIEW

**Comment by Baptistemontan** [comment]
That's awesome thanks a lot!

I wrote the docs pretty quickly to at least have some materials, but grammar is'nt my biggest strength, I'm not a native speaker either (I'm french, that's why everything is english/french haha). Any typos fixes are more than welcome.

> docs/book/src/declare/03_plurals.md:100

Line 111 you probably meant, the `0 <= count > 5`? Yes you're right, bad comparaison, thanks catching it.

File: docs/book/src/declare/03_plurals.md [hunk 1]
```diff
@@ -44,10 +44,11 @@ You can also declare a range where the translations is used:
 }
 ```
 
-You can use all Rust ranges syntax: `s..e`, `..e`, `s..`, `s..=e`, `..=e` or even `..` ( `..` will considered fallback `_`)
+You can use all Rust ranges syntax: `s..e`, `..e`, `s..`, `s..=e`, `..=e` or even `..` ( `..` will be considered fallback `_`)
 
 ## Number type
 
+// TODO: here its `i64` in usage/03_t_macro.html#plurals it says '(default is `i32`)'
```
Comment (Baptistemontan) [inline docs/book/src/declare/03_plurals.md:51]: YES! default is now `i32`, I missed that spot...

File: docs/book/src/declare/06_mix_kinds.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Mixing Kinds
 
-What happen if for a key you declare plurals in one locale, interpolation in another, and a simple string in a third ?
+What happens if for a key you declare plurals in one locale, interpolation in another, and a simple string in a third ?
 
-Well this is totally allowed, but you will still need to supply all values/component of every locales combined when using the translation, regardless of what the current locale is.
+Well this is totally allowed, but you will still need to supply all values/components of every locale combined when using the translation, regardless of what the current locale is.
 
-What is not allowed to mix is subkeys, the key must be subkeys in all locale.
+What is not allowed to mix are subkeys, the key must be a subkey in all locales.
```
Comment (Baptistemontan) [inline docs/book/src/declare/06_mix_kinds.md:7]: This is a difficult thing to phrase, my intent was to explain that if a key contain subkeys, that key must contain subkeys in all other locales, `"the key must be a subkey"` feels like the key being one depth deeper.

**Comment by Baptistemontan** [comment]
You really must have done a global search for `exemple`, I always mistake it with the french translation

File: docs/book/src/declare/06_mix_kinds.md (hunk 2 above)
Comment (AD4MANTIS) [reply docs/book/src/declare/06_mix_kinds.md:7]: How about we change it to something like this:
`"What is not allowed to mix are subkeys. If a key has subkeys in one locale, the key must have subkeys in all locales."`

Comment (Baptistemontan) [reply docs/book/src/declare/06_mix_kinds.md:7]: Perfect!

**Comment by Baptistemontan** [comment]
Thanks a lot!

**Pull request merged**